<b>Question:</b>
```
What piece of code loads the file 'data.pickle', which contains a dict object, into the variable data? You can assume that the directory containing 'data.pickle' is in your path (i.e., is accessible). 

*The end result should be that the variable data is a dict object.
```
```py
import pickle
with open('data.pickle') as f:
    data = pickle.load(f)
 
import pickle
with open('data.pickle') as f:
    data = f.open()
 
import pickle
data = pickle.open(f)
 
import pickle
with open('data.pickle') as f:
    data = f
```
<b>Working:</b><br>
<code>option_1.py</code>
```py
import pickle
with open('data.pickle') as f:
    data = pickle.load(f)
 ```
 <code>option_2.py</code>
 ```py
import pickle
with open('data.pickle') as f:
    data = f.open()
 ```
 <code>option_3.py</code>
 ```py
import pickle
data = pickle.open(f)
 ```
 <code>option_4.py</code>
 ```py
import pickle
with open('data.pickle') as f:
    data = f
```
<b>Execution along with Git:</b>
```py
dell@DELL3521 /d/WinPython-64bit-3.4.3.2
$ python option_1.py
Traceback (most recent call last):
  File "option_1.py", line 3, in <module>
    data=pickle.load(f)
TypeError: 'str' does not support the buffer interface

dell@DELL3521 /d/WinPython-64bit-3.4.3.2
$ python option_2.py
Traceback (most recent call last):
  File "option_2.py", line 3, in <module>
    data=f.open()
AttributeError: '_io.TextIOWrapper' object has no attribute 'open'

dell@DELL3521 /d/WinPython-64bit-3.4.3.2
$ python option_3.py
Traceback (most recent call last):
  File "option_3.py", line 2, in <module>
    data=pickle.open(f)
AttributeError: 'module' object has no attribute 'open'

dell@DELL3521 /d/WinPython-64bit-3.4.3.2
$ python option_4.py
$
```
But... this seems to work as correct answer
```py
with open('data.pickle', 'rb') as f:
```
Source: https://docs.python.org/3.1/library/pickle.html
